libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/MapboxSpeechPlayer.java

```diff
@@ -152,18 +152,23 @@ public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response)
         if (response.isSuccessful()) {
           executeInstructionTask(response.body());
         } else {
-          onError();
+          try {
+            onError(response.errorBody().string());
+          } catch (IOException exception) {
+            onError("");
+          }
         }
       }
 
       @Override
       public void onFailure(Call<ResponseBody> call, Throwable throwable) {
-        onError();
+        onError(throwable.getMessage());
       }
     });
   }
 
-  private void onError() {
+  private void onError(String error) {
+    Timber.e(error);
```

Comment:
Was wondering if we should add the error message to `InstructionListener#onError` and log when the callback is fired 🤔 What do you think?

Reply:
@Guardiola31337 so you're saying make `InstructionListener` an abstract class instead of an interface?